pkg/apis/core/v1/helper/helpers_test.go

```diff
@@ -792,3 +844,86 @@ func TestMatchNodeSelectorTerms(t *testing.T) {
 		})
 	}
 }
+
+func TestMatchTopologySelectorTerms(t *testing.T) {
+	type args struct {
+		topologySelectorTerms []v1.TopologySelectorTerm
+		nodeLabels            labels.Set
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "nil terms",
```

Comment:
All existing storageclasses are going to have `allowedTopologies == nil`.  Will this cause the selector to match nothing?

We either need:
* A top level `TopologySelector`, where if nil, it selects everything
* Select everything if `len(allowedTopologies)` is 0.  I double checked `NodeSelector` implementation, the `podMatchesNodeSelectorAndAffinityTerms` function treats nil/0 length `NodeSelectorTerms` as select nothing.  However, `ValidateNodeSelector` requires `NodeSelectorTerms > 0` . @k82cn can you clarify this discrepancy?

Reply:
oops, I ignored the case while removing the selector layer.
Will add extra logic in `MatchTopologySelectorTerms` for now, and revisit the part if it's agreed to add `TopologySelector` back.
Personally I think it looks better with the selector layer if we expect the structs behavior like that of NodeSelector